src/prometheus_alert_rules/memory.rules

```diff
@@ -0,0 +1,69 @@
+groups:
+- name: HostMemory
+  rules:
+  - record: node_memory_MemUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_MemFree_bytes
+          + node_memory_Cached_bytes
+          + node_memory_Buffers_bytes
+          + node_memory_SReclaimable_bytes
+        ) / node_memory_MemTotal_bytes
+      ))
+  - record: node_memory_SwapUsed_percentage
+    expr: |
+      100 * (1 - (
+        (
+          node_memory_SwapFree_bytes
+          + node_memory_SwapCached_bytes
+        ) / node_memory_SwapTotal_bytes
+      ))
+  - alert: HostMemoryFillsUp
+    expr: predict_linear(node_memory_MemUsed_percentage[30m], 5*60) >= 90
+    for: 2m
+    labels:
+      severity: warning
+    annotations:
+      summary: '[Prediction] Host memory usage will increase to {{ $value | printf "%.0f" }}% in the near future (instance {{ $labels.instance }})'
+      description: >-
+        Host can potentially reach {{ $value | printf "%.0f" }}% memory utilization and risk an OOM kill.
+          VALUE = {{ $value }}
+          LABELS = {{ $labels }}
+        The 5-minute-ahead prediction is made as a linear regression from the last 30 minutes of data.
+  - alert: HostMemoryFull
+    expr: avg_over_time(node_memory_MemUsed_percentage[1m]) > 95
+    for: 2m
+    labels:
+      severity: critical
+    annotations:
+      summary: Host memory usage reached {{ $value | printf "%.0f" }}% load (instance {{ $labels.instance }})
+      description: >-
+        Host memory usage reached {{ $value | printf "%.0f" }}%
+          VALUE = {{ $value }}
+          LABELS = {{ $labels }}
+  - alert: HostSwapFull
+    expr: |
+      avg_over_time(node_memory_MemUsed_percentage[1m]) > 90
```

Comment:
as above, 90% seems a bit low

Reply:
Even for swap memory 90% is not enough? I think that in most cases it should be more than enough.

Reply:
No you are right I forgot about this. In this rule we are checking memory and swap usage. Instead of checking only swap more than threshold I chose to check if memory usage is high enough (more than 90%) and swap usage is more than 50%. Does this make sense? I think that 90% for memory is not really important, since if swap usage grow -> memory usage should be high.

Reply:
I think this make sense. Let's try it out, and if there is still work to be done, we can fine-tune it in a subsequent PR.